vortex-ipc/src/codecs/array_reader/mod.rs

```diff
@@ -21,12 +24,18 @@ pub trait ArrayReader: Stream<Item = VortexResult<Array>> {
 
 /// An adapter for a stream of array chunks to implement an ArrayReader.
 #[pin_project]
-struct ArrayReaderAdapter<S> {
+pub struct ArrayReaderAdapter<S> {
     dtype: DType,
     #[pin]
     inner: S,
 }
 
+impl<S> ArrayReaderAdapter<S> {
```

Comment:
I would maybe group this with the next block? It's a bit pointless to be able to create one of these and not be able to do anything with it.